### DISCUSSION TITLE
OPA functionality for A/B Testing? "counting"

### DISCUSSION THREAD

**Discussion by hpvd** [open-policy-agent/community#427] [OPA and Rego] [answered]
We are thinking about using OPA for A/B-testing as decision-maker to decide which variant to use for a task
(in a classical frontend A/B tests, it would be to decide about which element to show to a user, but we are looking not only for FE testing...)

To start in the right direction:
=> Is there already an internal function in OPA to help in this use-case,
e.g. to make decisions, based on counting the decisions already done, when the goal would be to decide e.g.
- 30 decisions out of 100 for A and
- 70 out of 100 for B.

(Of course this is only the core and in practice there is additional data to be taken into account for the decision...)

Since I could not find any info on this, I appreciate any hint/advice!

**Comment by anderseknert** [comment, marked as answer]
Hi @hpvd 👋 

One challenge you'll need to deal with for that scenario is that OPA itself doesn't carry state between requests. So you can't increment some value internally and then read that in the context of the next request. What you could do is of course to let the client do the counting, and provide the number to OPA as part of the input of each request. OPA could then use that, and any other relevant data, to make decisions.

**Reply by hpvd** [reply]
@anderseknert Many thanks for this fast and detailed answer!
